src/app/client/src/app/modules/shared-feature/components/popup/popup.component.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { PopupComponent } from './popup.component';
+import { SuiModule } from 'ng2-semantic-ui';
+import { ActivatedRoute, Router } from '@angular/router';
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { OrgDetailsService, FrameworkService, FormService, LearnerService, CoreModule } from '@sunbird/core';
+import {ConfigService, ResourceService, Framework, ToasterService, SharedModule} from '@sunbird/shared';
+import {throwError as observableThrowError, of as observableOf,  Observable } from 'rxjs';
+import { Response } from './popup.component.spec.data';
+
+describe('PopupComponent', () => {
+  let component: PopupComponent;
+  let fixture: ComponentFixture<PopupComponent>;
+  const resourceBundle = {
+    'messages': {
+      'emsg': {
+        'm0005': 'Fetching data failed, please try again later...'
+      }
+    }
+  };
+  class RouterStub {
+    navigate = jasmine.createSpy('navigate');
+  }
+  const fakeActivatedRoute = {
+    'params': observableOf({ }),
+    'queryParams': observableOf({ })
+  };
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule, SharedModule.forRoot(), CoreModule.forRoot()],
+      declarations: [ PopupComponent ],
+      providers: [ConfigService,  LearnerService, OrgDetailsService, FrameworkService, FormService,
+        { provide: ResourceService, useValue: resourceBundle },
+        { provide: Router, useClass: RouterStub },
+        { provide: ActivatedRoute, useValue: fakeActivatedRoute }],
+        schemas: [NO_ERRORS_SCHEMA]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(PopupComponent);
+    component = fixture.componentInstance;
+  });
+  it('should call ngOninit when framework throw error', () => {
```

Comment:
test case statement is incorrect. why ngOnInit needs to be called when there is error in framework api response?